homeassistant/components/media_player/cmus.py

```diff
@@ -0,0 +1,223 @@
+"""
+Support for interacting with and controlling the cmus music player.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.mpd/
+"""
+
+import logging
+
+import eventlet
+
+from homeassistant.components.media_player import (
+    MEDIA_TYPE_MUSIC, MEDIA_TYPE_PLAYLIST, SUPPORT_NEXT_TRACK, SUPPORT_PAUSE,
+    SUPPORT_PREVIOUS_TRACK, SUPPORT_TURN_OFF, SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_SET, SUPPORT_PLAY_MEDIA, SUPPORT_SEEK,
+    MediaPlayerDevice)
+from homeassistant.const import (STATE_OFF, STATE_PAUSED, STATE_PLAYING,
+                                 CONF_HOST, CONF_NAME, CONF_PASSWORD,
+                                 CONF_PORT)
+
+_LOGGER = logging.getLogger(__name__)
+REQUIREMENTS = ['pycmus>=0.0.3']
+
+SUPPORT_CMUS = SUPPORT_PAUSE | SUPPORT_VOLUME_SET | SUPPORT_TURN_OFF |  \
+    SUPPORT_TURN_ON | SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | \
+    SUPPORT_PLAY_MEDIA | SUPPORT_SEEK
+
+
+def setup_platform(hass, config, add_devices, discover_info=None):
+    """Setup the Cmus platform."""
+    host = config.get(CONF_HOST, None)
+    password = config.get(CONF_PASSWORD, None)
+    port = config.get(CONF_PORT, None)
+    name = config.get(CONF_NAME, None)
+    if host and not password:
+        _LOGGER.error("A password must be set if using a remote cmus server")
+        return False
+    cmus_remote = CmusDevice(host, password, port, name)
+    add_devices([cmus_remote])
+
+
+class CmusDevice(MediaPlayerDevice):
+    """Representation of a running cmus."""
+
+    # pylint: disable=no-member, too-many-public-methods, abstract-method
+    def __init__(self, server, password, port, name):
+        """Initialize the CMUS device."""
+        remote = eventlet.import_patched('pycmus.remote')
+
+        if server:
+            port = port or 3000
+            self.cmus = remote.PyCmus(server=server, password=password,
+                                      port=port)
+            auto_name = "cmus-%s" % server
+        else:
+            self.cmus = remote.PyCmus()
+            auto_name = "cmus-local"
+        self._name = name or auto_name
+        self.status = {}
+        self.update()
+
+    def update(self):
+        """Get the latest data and update the state."""
+        status = self.cmus.status()
+        status_list = status.split('\n')
```

Comment:
It was mostly the poor order I ended up implementing things. I didn't think about this until I started working on the home-assistant patch. I just pushed a pycmus patch to add this functionality:

https://github.com/mtreinish/pycmus/commit/717b069db352944dc2784a9869b0f6e042da05ee

I'll push a release including this shortly and respin the patch to just use the new method